App/ViewModels/SettingsViewModel.cs

```diff
@@ -20,16 +20,24 @@ public partial class SettingsViewModel : ObservableObject
     [ObservableProperty]
     public partial bool StartOnLogin { get; set; }
 
-    private ISettingsManager _settingsManager;
+    private ISettingsManager<CoderConnectSettings> _connectSettingsManager;
+    private CoderConnectSettings _connectSettings = new CoderConnectSettings();
     private IStartupManager _startupManager;
 
-    public SettingsViewModel(ILogger<SettingsViewModel> logger, ISettingsManager settingsManager, IStartupManager startupManager)
+    public SettingsViewModel(ILogger<SettingsViewModel> logger, ISettingsManager<CoderConnectSettings> settingsManager, IStartupManager startupManager)
     {
-        _settingsManager = settingsManager;
+        _connectSettingsManager = settingsManager;
         _startupManager = startupManager;
         _logger = logger;
-        ConnectOnLaunch = _settingsManager.ConnectOnLaunch;
-        StartOnLogin = _settingsManager.StartOnLogin;
+        // Application settings are loaded on application startup,
+        // so we expect the settings to be available immediately.
+        var settingsCache = settingsManager.GetFromCache();
+        if (settingsCache is not null)
+        {
+            _connectSettings = settingsCache.Clone();
+        }
```

Comment:
When returning a cached value we should probably just always clone it from within ISettingsManager